youtube_dl/extractor/iol.py

```diff
@@ -0,0 +1,98 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+
+import re
+
+
+class IOLIE(InfoExtractor):
+    _VALID_URL = r'http://(tviplayer|(www\.tvi24))\.iol\.pt/.*/(?P<id>[0-9a-f]{24})($|\/)'
+    _TESTS = [{
+        'url': 'http://tviplayer.iol.pt/programa/euromilhoes/53c6b3153004dc006243b07b/video/55f878f90cf203f8b03cea6d',
+        'md5': '983ca0edae370af67c458c6e5a71aae5',
+        'info_dict': {
+            'id': '55f878f90cf203f8b03cea6d',
+            'ext': 'mp4',
+            'title': u'Euromilhões - 15 de setembro de 2015',
+            # 'tumbnail': 'http://www.iol.pt/multimedia/oratvi/multimedia/imagem/id/55f87a280cf2e6961770f01f',
+            'description': u'Com Mónica Jardim'
+        }
+    }, {
+        'url': 'http://tviplayer.iol.pt/programa/isso-e-tudo-muito-bonito-mas/55f30f2e0cf2a6b037fc1f2f/video/55f730c40cf23fa665481b18',
+        'md5': 'ef5171a5abf69197726e5d7c7633c27a',
+        'info_dict': {
+            'id': '55f730c40cf23fa665481b18',
+            'ext': 'mp4',
+            'title': u'Isso é tudo muito bonito, mas: Concatena, filho, concatena',
+            # 'tumbnail': 'http://www.iol.pt/multimedia/oratvi/multimedia/imagem/id/55f737330cf23fa665481b2d',
+            'description': u'Quando os adversários unem perspectivas.'
+        }
+    }, {
+        'url': 'http://www.tvi24.iol.pt/videos/passos-criacao-de-emprego-e-facto-muito-importante/55f816640cf2e6961770ef7a/2',
+        'md5': 'd836f1225c289c7987beddebe11619b9',
+        'info_dict': {
+            'id': '55f816640cf2e6961770ef7a',
+            'ext': 'mp4',
+            'title': u'Passos: criação de emprego é facto muito importante',
+            # 'tumbnail': 'http://www.iol.pt/multimedia/oratvi/multimedia/imagem/id/55f8180e0cf21413dfb1d96d/',
+            'description': u'PM sublinha que é o nível mais elevado de há vários anos'
+        }
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        title = self._og_search_title(webpage)
+        title = re.sub(r' \| TVI Player$', '', title, re.IGNORECASE)
```

Comment:
doesn't always contain `| TVI Player`.
can be extracted from the ioplayer title.
for example:
```javascript
.iolplayer({
					autostart: true,
                    videoUrl: 'http://vod-auth.iol.pt/vod/_definst_/smil:55f730c40cf23fa665481b18-L/playlist.m3u8?wmsAuthSign=c2VydmVyX3RpbWU9MTAvMTAvMjAxNSA4OjI5OjIzIEFNJmhhc2hfdmFsdWU9Y1FlYVJYWEovMWR6Z1pUNTllOWJidz09JnZhbGlkbWludXRlcz0xNDQwJmlkPTBmYzE2NGVmLTQ3OTktNDY3Mi1iZTI2LTcyN2FmMzM4NGRhOA==',
                    imageId: '55f737330cf23fa665481b2d',
                    title: 'Isso é tudo muito bonito, mas: Concatena, filho, concatena',
```